frontend/openchat-service-worker/src/service_worker.ts

```diff
@@ -161,20 +166,19 @@ async function handleNotificationClick(event: NotificationEvent): Promise<void>
 
     if (windowClients.length > 0) {
         const window = windowClients[0];
-        window.focus();
-
         window.postMessage({
             type: "NOTIFICATION_CLICKED",
             path: event.notification.data.path,
         });
+        await window.focus();
```

Comment:
I'm hoping this will fix the issue @megrogan was facing where it wouldn't focus the tab after clicking a notification